### ISSUE TITLE
Cloning of response leads to issues

### ISSUE THREAD

**Issue by benmccann** [sveltejs/kit#591]
This line:

https://github.com/sveltejs/kit/blob/8805c6daa6b11a46d2d373c3db890227b27e660d/packages/kit/src/runtime/server/page.js#L128

Causes issues in Node where it dies unexpectedly without any message on server. The Node high water mark limit is much smaller than browser. Can increase high water mark as workaround

There's multiple issues talking about this in `node-fetch` (e.g. [671](https://github.com/node-fetch/node-fetch/pull/671) and [151](https://github.com/node-fetch/node-fetch/issues/151)). Their recommendation is to stream and provide back pressure: https://github.com/node-fetch/node-fetch/issues/151#issuecomment-482777213

**Comment by Rich-Harris** [comment]
If someone can create a repro it would help a lot

**Comment by benmccann** [comment]
There was a comment on Discord that they're only reproducing this if the request goes through Nginx: https://discord.com/channels/457912077277855764/819723698415599626/823759556789272597

**Comment by stalkerg** [comment]
@Rich-Harris yes, self-locking happens here https://github.com/sveltejs/kit/blob/master/packages/kit/src/runtime/server/page.js#L141 

If an HTTP packet contains more than two TCP packages (the first packet is usually the header) and one of these TCP packages for the body did not fit into the write buffer.
Basically, `await clone.text()` waiting for a response from the stream, but this stream waiting drying (reading) buffer by original `response` what can't be returned from the function because we waiting `await clone.text()`.

Basically, `const clone = response.clone();` can work only if all tcp package for HTTP body fit into write buffer OR JS interpretation/compiler/VM smart enough to detect self-locks what NodeJS not.

**Comment by Rich-Harris** [comment]
Is there any way to interpret this other than '`fetch` is fundamentally broken'?

**Comment by stalkerg** [comment]
If you talking about sveltekit fetch - then yes. 
PS one more link https://github.com/whatwg/streams/issues/506#issuecomment-242073695

**Comment by benmccann** [comment]
The way to fix this problem is to resolve the cloned response in parallel with the original. For example:
```
const fetch = require('node-fetch');

const response = await fetch('https://example.com');
const r1 = await response.clone();

const results = await Promise.all([response.json(), r1.text()]);
```

The response in SvelteKit is used in two places:
1. it's serialized and put into the page for hydration
2. it's used in the server-side implementation of `load` as desired by the user

The problem here is that we're trying to do these basically in serial by basically intercepting the response.

I can't think of a way to read the stream in the two places in parallel while maintaining our current API. The issue is that `load` takes `fetch` as a parameter, so we can't call `load` and serialize the response in parallel.

So as much as I hate to say it, I think that `SvelteKit's` `load` implementation is fundamentally broken and I don't think there's a way to fix it while keeping our current API. The only thing I can really think at this point is that we probably made the wrong decision in https://github.com/sveltejs/kit/issues/268 and should go back to a Sapper-like implementation where the user can handle the `fetch` response however they like and then they hand it off to us. At that point we can take the user's response and **in parallel** render the Svelte components and serialize the data for hydration

This is why we're in beta and not final release :grin:

**Comment by Rich-Harris** [comment]
> So as much as I hate to say it, I think that `SvelteKit's` `load` implementation is fundamentally broken and I don't think there's a way to fix it while keeping our current API

o ye of little faith #635

**Comment by Conduitry** [comment]
Proxies are truly the Spanish Inquisition of js language features.

**Comment by nosovk** [comment]
Could it be that moving to https://undici.nodejs.org/#/?id=undicifetchinput-init-promise can help to fix an issue (and yes undici is not equal to node-fetch)

**Comment by stalkerg** [comment]
@nosovk probably can't, because it's very fundamental issue, but current proxys really work
